pkg/osquery/enrollment_details.go

```diff
@@ -0,0 +1,140 @@
+package osquery
+
+import (
+	"bytes"
+	"context"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"os"
+	"runtime"
+	"time"
+
+	"github.com/kolide/kit/version"
+	"github.com/kolide/launcher/pkg/agent"
+	"github.com/kolide/launcher/pkg/osquery/runsimple"
+	"github.com/kolide/launcher/pkg/service"
+	"github.com/pkg/errors"
+)
+
+func getEnrollDetails(osquerydPath string) (service.EnrollmentDetails, error) {
+	var details service.EnrollmentDetails
+
+	// To facilitate manual testing around missing enrollment details,
+	// there is a environmental variable to trigger the failure condition
+	if os.Getenv("LAUNCHER_DEBUG_ENROLL_DETAILS_ERROR") == "true" {
+		return details, errors.New("Skipping enrollment details")
+	}
+
+	// If the binary doesn't exist, bail out early.
+	if info, err := os.Stat(osquerydPath); os.IsNotExist(err) {
+		return details, fmt.Errorf("no binary at %s", osquerydPath)
+	} else if info.IsDir() {
+		return details, fmt.Errorf("%s is a directory", osquerydPath)
+	}
+
+	query := `
+	SELECT
+		osquery_info.version as osquery_version,
+		os_version.build as os_build,
+		os_version.name as os_name,
+		os_version.platform as os_platform,
+		os_version.platform_like as os_platform_like,
+		os_version.version as os_version,
+		system_info.hardware_model,
+		system_info.hardware_serial,
+		system_info.hardware_vendor,
+		system_info.hostname,
+		system_info.uuid as hardware_uuid
+	FROM
+		os_version,
+		system_info,
+		osquery_info;
+
+`
+
+	var respBytes bytes.Buffer
+
+	osq, err := runsimple.NewOsqueryProcess(
+		osquerydPath,
+		runsimple.RunSql([]byte(query)),
+		runsimple.WithStdout(&respBytes),
+	)
+	if err != nil {
+		return details, fmt.Errorf("create osquery for enrollment details: %w", err)
+	}
+
+	osqCtx, osqCancel := context.WithTimeout(context.TODO(), 5*time.Second)
```

Comment:
This function is called by `Enroll` which has access to a preexisting `ctx` -- do we want to pass that in to this function and use it here instead of `context.TODO()`?

I saw later in the simple runner you left a comment that osquery should have its own context -- why?

Reply:
Two answers. 

Good point -- yes, this should use the context. I've updated it.

The later comment is about running osquery as a subprocess. If we wanted it to run longer, it might want it's own context so we can separate it and manage the shutdown more cleanly. Not totally sure. <handwave about rungroup semantics>